README.md

```diff
@@ -33,59 +33,131 @@ struct Velocity
 	{
-		*pos.x += *vel.x
-		*pos.y += *vel.y
+		// ask the query for components
+		Position* pos;
+		Velocity* vel;
+
+		AliasSeq!(pos, vel) = query.get!(Position, Velocity)(entity);
 	}
+
+	// loop reversed
+	foreach_reverse(entity; query) { /*...*/ }
+	foreach_reverse(entity, pos, vel; query.each()) { /*...*/ }
+
+	// ask the query for entites in it
+	assert(query.contains(Entity(0)));
 }
 
-// loops through all (Position, Velocity) components and entities with (Grounder)
-void systemB(Query!(Tuple!(Entity,Position,Velocity), With!Grounder) query)
+void main()
 {
-	foreach(e, pos, vel; query)
+	auto world = new EntityManager();
+
+	foreach (i; 0 .. 10) with (world)
 	{
-		*pos.x += *vel.x
-		*pos.y += *vel.y
+		if (!(i & 1))
+			entity(Entity(i)) // constructs or gets the entity with id 'i'
+				.emplace!Position(i, i)
+				.emplace!Velocity(i * 0.1f, i * 0.1f);
 	}
+
+	system(world.query!(Position, Velocity));
 }
+```
+
+## Query Rules
+```d
+// Select:  defines which components are selected, and filters entities that own them
+// With:    behaves as Select but the components are not returned
+// Without: filters entities that do not own the components
+
+import vecs;
+
+// queries entities with A, B, C, ...
+alias Q1 = EntityManager.Query!(A, B, C, ...);
+
+// queries entities with A, B, C but Selects only A, B
+alias Q2 = EntityManager.Query!(A, B).With!C;
+
+// queries entities with A, B and without C but Selects only A
+alias Q3 = EntityManager.Query!A.With!B.Without!C;
 
-// loops through all (Position, Velocity) components of entities with (Grounder) and without (Crawler)
-void systemC(Query!(Tuple!(Position,Velocity), Tuple!(With!Grounder, Without!Crawler)) query)
+// the above alias to:
+alias _Q1 = Query!(EntityManager, Select!(A, B, C, ...));
+alias _Q2 = Query!(EntityManager, Select!(A, B), With!C);
+alias _Q2 = Query!(EntityManager, Select!A, With!B, Without!C);
+
+void main()
 {
-	...
+	auto world = new EntityManager();
+
+	// when using rules the Select wrapper must be used
+	world.query!(Select!(A, B), With!(C, D), ...);
+
+	// Select is a special Rule and can only be used as the first argument
+	// All other argument must be QueryRules 'With and Without'
+	static assert(!_traits(compiles, world.query!(Select!A, Select!B)));
 }
+```
+
+## Safety
+EntityManagerT can be used in `@safe pure nothrow` code and some parts are
+`@nogc`. What defines the majority of code safetyness is the Signal. Right now
+in D it is impossible to store functions with different attributes in one common
+type and still be able to track them. The common attribute to all functions is
+having none, which is the same as having `@system`. This forces all code that
+depends on it to be `@system` as well (which is almost the entire lib)
+
+The user can define their own safety restriction by passing a delegate with the
+attributes they desire for their callbacks. Those attributes influence the
+safetyness of almost the entire lib as explained above.
```

Comment:
If I understood correctly, you refer to the code below, not above:
```suggestion
safetyness of almost the entire lib as explained below:
```

Reply:
No, it refers to the first paragraph. The code example demonstrates that sentence, and the last part of the same is referring to the explanation given in the previous paragraph.